include/deal.II/base/vectorization.h

```diff
@@ -175,8 +175,23 @@ class VectorizedArray
    */
   static const unsigned int n_array_elements = 1;
 
-  // POD means that there should be no user-defined constructors, destructors
-  // and copy functions (the standard is somewhat relaxed in C++2011, though).
+  // POD means that there should be no user-defined copy constructors or
+  // destructors (the standard is somewhat relaxed in C++2011, though). We need
+  // to define default constructor and conversion constructor (from Number) in
+  // order to properly and idiomatically construct
+  // Tensor<rank,dim,VectorizedArray<Number>>
+
+  /**
+   * Default constructor
+   */
+  VectorizedArray() = default;
+
+  /**
+   * Conversion constructor from scalar
+   */
+  explicit constexpr VectorizedArray(const Number scalar)
+    : data{scalar}
+  {}
```

Comment:
Have you checked what `std::is_pod<VectorizedArray<double>>::value` returns after this change, both for this default class and the SIMD classes below? If this does not succeed, we could probably relax the 'no constructors' design of this class anyway, because it would simplify code in a number of places. It would be good if we could at least guarantee `std::is_standard_layout` as that assumption is used implicitly all over the place (not that I would have seen any sane compiler doing it wrong also without that yet). A follow-up question would be to what extent we could make this class `constexpr` in a similar vein as `Tensor`, but that can wait for another PR for sure.

Reply:
I removed unnecessary constructors from VectorizedArray<Number>, but these constructors are still defined for the specializations of the class. I added static_assert for make sure the class has standard layout.